model_analyzer/device/gpu_device_factory.py

```diff
@@ -223,19 +226,28 @@ def get_cuda_visible_gpus(self):
             UUIDs of the DCGM supported devices visible to CUDA
         """
 
-        cuda_visible_gpus = []
-        if numba.cuda.is_available():
-            for cuda_device in numba.cuda.list_devices():
-                try:
-                    cuda_visible_gpus.append(
-                        self.get_device_by_cuda_index(cuda_device.id))
-                except TritonModelAnalyzerException:
-                    # Device not supported by DCGM, log warning
-                    logger.warning(
-                        f"Device '{str(cuda_device.name, encoding='ascii')}' with "
-                        f"cuda device id {cuda_device.id} is not supported by DCGM."
-                    )
-        return cuda_visible_gpus
+        nvmlInit()
+
+        gpu_devices = []
+        try:
+            devices = nvmlDeviceGetCount()
+            for device_id in range(devices):
```

Comment:
I noticed the previous variable name was `cuda_visible_gpus`, so just wanted to mention this:
 
Might not be an issue for anyone, but `nvml` doesn't respect `CUDA_VISIBLE_DEVICES`. So if you have any tests that exploit this env var for isolation inside the container, it likely won't work.

Isolating gpus in a container through `docker run --gpus '"device=1,3,5"' ...` will work as expected though.

Reply:
the function name is get_cuda_visible_gpus, so we should honor that. Is there a way to do so? Can we read that env variable here to filter the gpus?

Reply:
Ryan, thanks for looking at this! I've added in logic to check for CUDA_VISIBLE_DEVICES and map them to the device ids.